File: Example/DramAdsObjcExample/DramAdsObjcExample/ViewConrollers/Ads/Ads/Rewarded/RewardedViewController.m
//
//  RewardedViewController.m
//  DramAdsObjcExample
//
//  Created by Khoren Asatryan on 23.09.23.
//

#import "RewardedViewController.h"
#import <DramAds/DramAds-Swift.h>

@interface RewardedViewController () <DMRewardedAdDelegate, DMRewardedAdUIDataSource>

@end

@implementation RewardedViewController

- (IBAction)didSelectLoadAdButton:(id)sender {
    
    //a23c4147ab4d03ecdb937b15a3c3dc60 2
    //c418065bb67183661a1249cbdebdf071 1
    
    __weak RewardedViewController *weakSelf = self;
    [DMRewardedAd loadAdWithPlacementKey:@"c418065bb67183661a1249cbdebdf071" success:^(DMRewardedAd * ad) {
        
        ad.delegate = weakSelf;
        ad.uiDataSource = weakSelf;
        [ad showIn:self];
        
    } failure:^(DMServiceAdError * error) {
        NSLog(@"%@", error.message);
    }];
 
}

- (void)viewDidLoad {
    [super viewDidLoad];
    
}

#pragma mark - DMRewardedAdDelegate

- (void)rewardedAdDidStartPlaying:(DMRewardedAd *)ad {
    NSLog(@"rewardedAd DidStartPlaying %@", ad.placementKey);
}

- (void)rewardedAdDidClicked:(DMRewardedAd *)ad {
    NSLog(@"rewardedAd AdDidClicked %@", ad.placementKey);
}

- (void)rewardedAdDidCompleted:(DMRewardedAd *)ad error:(DMServiceAdError *)error {
    NSLog(@"rewardedAd AdDidCompleted %@", ad.placementKey);
}

- (void)rewardedAdDidSendImpression:(DMRewardedAd *)ad error:(DMServiceAdError *)error {
    NSLog(@"rewardedAd AdDidSendImpression %@", ad.placementKey);
}

#pragma mark - DMRewardedAdUIDataSource

- (UIInterfaceOrientationMask)rewardedAdSupportedInterfaceOrientations:(DMRewardedAd *)ad {
    return self.supportedInterfaceOrientations;
}

- (UIInterfaceOrientation)rewardedAdPreferredInterfaceOrientationForPresentation:(DMRewardedAd *)ad {
    return self.preferredInterfaceOrientationForPresentation;
}

@end

File: Example/DramAds/Ads/RewardedViewController.swift
//
//  RewardedViewController.swift
//  DramAdsIosSdkExample
//
//  Created by Khoren Asatryan on 09.09.23.
//

import UIKit
import DramAds

class RewardedViewController: UIViewController {

    @IBAction private func didLoadAd(_ sender: Any) {
        self.dismiss(animated: true)
        
        //a23c4147ab4d03ecdb937b15a3c3dc60 2
        //c418065bb67183661a1249cbdebdf071 1
        
        DM.RewardedAd.load(placementKey: "c418065bb67183661a1249cbdebdf071") { [weak self] result in
            switch result {
            case .success(let ad):
                self?.show(ad: ad)
                break
            case .failure(let error):
                print(error as NSError)
                print("/////////////////")
                break
            }
        }
    
    }
    
    func show(ad: DM.RewardedAd) {
        ad.delegate = self
        ad.uiDataSource = self
        ad.show(in: self)
    }
    
}

extension RewardedViewController: DMRewardedAdDelegate {
    
    func rewardedAd(didStartPlaying ad: DM.RewardedAd) {
        print("rewardedAd didStartPlaying", ad.placementId)
    }
    
    func rewardedAd(didSendImpression ad: DM.RewardedAd, error: IDMError?) {
        print("rewardedAd didSendImpression", ad.placementId)
    }
    
    func rewardedAd(didClicked ad: DM.RewardedAd) {
        print("rewardedAd adDidClicked", ad.placementId)
    }
    
    func rewardedAd(didCompleted ad: DM.RewardedAd, error: IDMError?) {
        print("rewardedAd didCompleted", ad.placementId)
    }
    
}

extension RewardedViewController: DMRewardedAdUIDataSource {
    
    func rewardedAd(supportedInterfaceOrientations ad: DM.RewardedAd) -> UIInterfaceOrientationMask? {
        return .portrait
    }
    
}
